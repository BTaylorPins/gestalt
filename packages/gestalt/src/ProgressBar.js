// @flow strict
import { type Node } from 'react';
import Box from './Box.js';
import Flex from './Flex.js';
import Text from './Text.js';

type Props = {|
  /**
   * Prop description.
   *
   * Link: https://gestalt.pinterest.systems/progressbar#prop
   */
  accessibilityLabel?: string,
  /**
   * The header text for the component.
   */
  title: string,
  /**
   * The header text for the component.
   */
  helperText: string,
  /**
   * The datapoint value (e.g., 1.23M)
   */
  value: number,
|};

/**
 * [ProgressBar] https://gestalt.pinterest.systems/progressbar component should be used for ... on the page.
 * ![ProgressBar light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/ProgressBar.spec.mjs-snapshots/ProgressBar-chromium-darwin.png)
 * ![ProgressBar dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/ProgressBar-dark.spec.mjs-snapshots/ProgressBar-dark-chromium-darwin.png)
 */
export default function ProgressBar({
  accessibilityLabel,
  helperText,
  title,
  value = 0,
}: Props): Node {
  return (
    <Box width="100%">
      <Flex>
        <Flex.Item flex="grow">
          <Text size="200">{title}</Text>
        </Flex.Item>
        <Flex.Item>
          <Text size="200">{`${value * 100}%`}</Text>
        </Flex.Item>
      </Flex>
      <Box
        marginTop={2}
        marginBottom={2}
        aria-label={accessibilityLabel}
        color="secondary"
        width="100%"
        rounding="pill"
        height={16}
      >
        <Box color="successBase" width={`${value * 100}%`} rounding="pill" height={16} />
      </Box>
      <Text size="100" color="subtle">
        {helperText}
      </Text>
    </Box>
  );
}
