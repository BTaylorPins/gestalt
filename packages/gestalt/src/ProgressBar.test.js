// @flow strict
import { create } from 'react-test-renderer';
import ProgressBar from './ProgressBar.js';

describe('ProgressBar', () => {
  it('renders', () => {
    const tree = create(<ProgressBar name="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an accessibility label', () => {
    const tree = create(<ProgressBar accessibilityLabel="Test Accessibility Label" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
