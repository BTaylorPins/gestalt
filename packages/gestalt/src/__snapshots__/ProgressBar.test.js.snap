// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders 1`] = `
<div
  className="box"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap0 xsDirectionRow"
  >
    <div
      className="FlexItem flexGrow"
    >
      <div
        className="Text fontSize200 defaultText alignStart breakWord fontWeightNormal"
      />
    </div>
    <div
      className="FlexItem"
    >
      <div
        className="Text fontSize200 defaultText alignStart breakWord fontWeightNormal"
      >
        0%
      </div>
    </div>
  </div>
  <div
    className="box marginBottom2 marginTop2 pill secondary"
    style={
      Object {
        "height": 16,
        "width": "100%",
      }
    }
  >
    <div
      className="box pill successBase"
      style={
        Object {
          "height": 16,
          "width": "0%",
        }
      }
    />
  </div>
  <div
    className="Text fontSize100 subtleText alignStart breakWord fontWeightNormal"
  />
</div>
`;

exports[`ProgressBar renders an accessibility label 1`] = `
<div
  className="box"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap0 xsDirectionRow"
  >
    <div
      className="FlexItem flexGrow"
    >
      <div
        className="Text fontSize200 defaultText alignStart breakWord fontWeightNormal"
      />
    </div>
    <div
      className="FlexItem"
    >
      <div
        className="Text fontSize200 defaultText alignStart breakWord fontWeightNormal"
      >
        0%
      </div>
    </div>
  </div>
  <div
    aria-label="Test Accessibility Label"
    className="box marginBottom2 marginTop2 pill secondary"
    style={
      Object {
        "height": 16,
        "width": "100%",
      }
    }
  >
    <div
      className="box pill successBase"
      style={
        Object {
          "height": 16,
          "width": "0%",
        }
      }
    />
  </div>
  <div
    className="Text fontSize100 subtleText alignStart breakWord fontWeightNormal"
  />
</div>
`;
